.github/workflows/pages.yml

```diff
@@ -0,0 +1,46 @@
+name: GitHub Pages
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [16.x]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Install Hugo
+      uses: peaceiris/actions-hugo@v2
+      with:
+        hugo-version: 'latest'
+        extended: true
+
+    - name: Check install Hugo
+      run: hugo version
+
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Delete temporary directories
+      run: npm run clean
+
+    - name: Build production website
+      run: npm run build
+
+    - name: Deploy to GitHub Pages
+      uses: peaceiris/actions-gh-pages@v3
```

Comment:
I guess I'll need to go replace this with something that deploys to the main site? I can probably deploy to aegisub.github.io as well automatically.

Reply:
Maybe Github page + [custom domain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site)?


Or [Deploy website with Github Action](https://1082.xyz/deploy-website-to-server-using-github-actions/)